Author: José Manuel Santamaría Lema <panfaust@gmail.com>
Description: Revert "Fix potential error when identity path is a string (bad conversion)."
 This patch reverts this upstream commit:
 https://projects.kde.org/projects/kde/kdepimlibs/repository/revisions/33acd1d6
 .
 This commit should be reverted if we are using kmail 4.4.10, because it makes
 kmail forget the identity's settings of draft, templates and sent folders.
 -
 This patch must be removed if kdepimlibs are used with kmail 2, based on akonadi.

--- a/kpimidentities/identity.cpp
+++ b/kpimidentities/identity.cpp
@@ -428,20 +428,17 @@ QString Identity::dictionary() const
 
 QString Identity::templates() const
 {
-  const QString str = property( QLatin1String( s_templates ) ).toString();
-  return verifyAkonadiId(str);
+  return property( QLatin1String( s_templates ) ).toString();
 }
 
 QString Identity::drafts() const
 {
-  const QString str = property( QLatin1String( s_drafts ) ).toString();
-  return verifyAkonadiId(str);
+  return property( QLatin1String( s_drafts ) ).toString();
 }
 
 QString Identity::fcc() const
 {
-  const QString str = property( QLatin1String( s_fcc ) ).toString();
-  return verifyAkonadiId(str);
+  return property( QLatin1String( s_fcc ) ).toString();
 }
 
 QString Identity::transport() const
@@ -644,17 +641,3 @@ bool Identity::matchesEmailAddress( const QString & addr ) const
 
   return false;
 }
-
-QString Identity::verifyAkonadiId(const QString& str) const
-{
-  if(str.isEmpty())
-    return str;
-  bool ok = false;
-  const qlonglong val = str.toLongLong(&ok);
-  Q_UNUSED(val);
-  if(ok) {
-    return str;
-  } else {
-    return QString();
-  }
-}
diff --git a/kpimidentities/identity.h b/kpimidentities/identity.h
index 349ad4d..22b1f42 100644
--- a/kpimidentities/identity.h
+++ b/kpimidentities/identity.h
@@ -311,10 +311,6 @@ class KPIMIDENTITIES_EXPORT Identity
       void setUoid( uint aUoid );
 
     protected:
-      /** during migration when it failed it can be a string => not a qlonglong akonadi::id => fix it*/
-      /** remove it in kde5 */
-      QString verifyAkonadiId(const QString& str) const;
-
       /** @return true if the signature is read from the output of a command */
       bool signatureIsCommand() const;
 
